specification/botservice/resource-manager/Microsoft.BotService/stable/2017-12-01/botservice.json

```diff
@@ -349,28 +349,64 @@
         }
       }
     },
-    "/subscriptions/{subscriptionId}/providers/Microsoft.BotService/checkNameAvailability": {
-      "get": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.BotService/botServices/{resourceName}/channels/{channelName}": {
+      "put": {
         "tags": [
-          "BotService"
+          "Channel"
         ],
-        "description": "Check whether a bot name is available.",
-        "operationId": "BotServices_GetCheckNameAvailability",
+        "description": "Creates a Channel registration for a Bot Service",
+        "operationId": "Channels_Create",
         "x-ms-examples": {
-          "List Bots by Subscription": {
-            "$ref": "./examples/CheckNameAvailability.json"
+          "Create Bot": {
+            "$ref": "./examples/PutChannel.json"
           }
         },
         "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group within the user's subscription."
+          },
+          {
+            "name": "resourceName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "maxLength": 64,
+            "minLength": 2,
+            "pattern": "^[a-zA-Z0-9][a-zA-Z0-9_.-]*$",
+            "description": "The name of the Bot resource."
+          },

```

Comment:
The above are usually defined as a global parameters with the [`"x-ms-parameter-location":"method"`](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-parameter-location) property set.

This isn't strictly necessary, but it should lead to easier maintenance in the future for you. :)